thunder/torch/__init__.py

```diff
@@ -5425,15 +5425,16 @@ def register_default_torch_ops():
 def register_default_torch_op(torchfn: Callable, torch_module):
     fn_meta = meta_adaptor(torchfn)
     _fn = langctx(Languages.TORCH)(fn_meta)
+    torchfn_name = torchfn.__name__.split("_")[1] if torch_module == torch.special else torchfn.__name__
```

Comment:
Would you add a comment explaining what's happening here and why torch.special has special behavior?

Does this only happen for the torch.special submodule, and not other submodules like torch.linalg or torch.fft?

Reply:
This seems to originate with the C++ names in native_func.yaml (and then undone in the torch.foo Python modules themselves) and seems to affect quite a few functions / modules:
```
     42   python_module: fft
     98   python_module: linalg
      1   python_module: nested
     14   python_module: sparse
    170   python_module: special
```
Not mangled is the earliest
```
    286   python_module: nn
```
Maybe making a hypothesis and asserting that we find the function as `getattr(torch_module, torchfn_name)` would be cool.

Reply:
Hi @mruberry @t-vi , I added a function to infer the torch function name based on the torch module and function, then validate it using `getattr` as suggested
Additionally, since torch.linalg and torch.fft are mentioned here, I added them in the auto registration and tests